frontend/packages/process-editor/src/components/ConfigPanel/ConfigContent/EditLayoutSetName/EditLayoutSetName.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { StudioToggleableTextfield } from '@studio/components';
+import { KeyVerticalIcon } from '@studio/icons';
+import { useBpmnApiContext } from '../../../../contexts/BpmnApiContext';
+import { Paragraph } from '@digdir/designsystemet-react';
+import { useValidateLayoutSetName } from 'app-shared/hooks/useValidateLayoutSetName';
+import { useBpmnContext } from '../../../../contexts/BpmnContext';
+
+interface EditLayoutSetNameProps {
+  existingLayoutSetName: string;
+}
+export const EditLayoutSetName = ({
+  existingLayoutSetName,
+}: EditLayoutSetNameProps): React.ReactElement => {
+  const { t } = useTranslation();
+  const { bpmnDetails } = useBpmnContext();
+  const { layoutSets, mutateLayoutSetId } = useBpmnApiContext();
+  const { validateLayoutSetName } = useValidateLayoutSetName();
+
+  const handleOnLayoutSetNameBlur = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    const newName = event.target.value;
+    if (newName === existingLayoutSetName) return;
+    mutateLayoutSetId({ layoutSetIdToUpdate: existingLayoutSetName, newLayoutSetId: newName });
+  };
+
+  return (
+    <StudioToggleableTextfield
+      customValidation={(newLayoutSetName: string) =>
+        validateLayoutSetName(newLayoutSetName, layoutSets, existingLayoutSetName)
+      }
+      inputProps={{
+        icon: <KeyVerticalIcon />,
+        label: t('process_editor.configuration_panel_layout_set_name_label'),
+        value: existingLayoutSetName,
+        onBlur: (event) => handleOnLayoutSetNameBlur(event),
```

Comment:
I think being able to press enter to save the value would be nice here. Maybe altering the compoent to use form semantics / `onSubmit` makes sense here?

Reply:
Agree, but maybe we should do that for the StudioTextfield component in a separate issue? 🤔

Reply:
I agree, but maybe for `StudioToggleableTextfield` instead, as `StudioTextfield` might be used for non-form things?

Reply:
Good point! I will create an issue for it 💯